prelude/bootstrap.js

```diff
@@ -1051,8 +1079,7 @@ function payloadFileSync(pointer) {
 
   function statFromSnapshot(path_, cb) {
     var cb2 = cb || rethrow;
-    var path = normalizePath(path_);
-    // console.log("statFromSnapshot", path);
+    var path = normalizePathAndFollowLink(path_);
```

Comment:
Just to be sure, is it ok to follow so? Was the comment you added here wrong?

Reply:
Suppose that we have a path that is also a symlink.
When we asked for stat, shall we get the stat from the real file or for the link itself.
I ask myself this question. At one point I though we would need to separate the stat  from the link file from the real file. but it appears that we can avoid duplicating this information.
Looking at what the `fs.statSync` method does on a symlinked file we get the size of the realfile, the only difference is the nlink property that is set to 1.

![image](https://user-images.githubusercontent.com/680220/113001403-14217100-9171-11eb-9160-9fe247d540e9.png)
> fs.statSync("a1.js")
Stats {
  dev: 2890017917,
  mode: 33206,
  nlink: 1,
  uid: 0,
  gid: 0,
  rdev: 0,
  blksize: 4096,
  ino: 4503599628273627,
  size: 1729,
  blocks: 8,
  atimeMs: 1616854148464.44,
  mtimeMs: 1616503529688.595,
  ctimeMs: 1616503529688.595,
  birthtimeMs: 1616503255624.1414,
  atime: 2021-03-27T14:09:08.464Z,
  mtime: 2021-03-23T12:45:29.689Z,
  ctime: 2021-03-23T12:45:29.689Z,
  birthtime: 2021-03-23T12:40:55.624Z
}
> fs.statSync("a.js")
Stats {
  dev: 2890017917,
  mode: 33206,
  nlink: 1,
  uid: 0,
  gid: 0,
  rdev: 0,
  blksize: 4096,
  ino: 4503599628273627,
  size: 1729,
  blocks: 8,
  atimeMs: 1616854148464.44,
  mtimeMs: 1616503529688.595,
  ctimeMs: 1616503529688.595,
  birthtimeMs: 1616503255624.1414,
  atime: 2021-03-27T14:09:08.464Z,
  mtime: 2021-03-23T12:45:29.689Z,
  ctime: 2021-03-23T12:45:29.689Z,
  birthtime: 2021-03-23T12:40:55.624Z
}

so using the stat stored in the real file snap safe some bytes as we emulate the behavior of fs.

So yes it is ok to resolve the link here,
(the comment was out dated)

Reply:
thanks for the explanation :) Resolved